x-pack/plugins/security_solution/public/common/containers/source/use_data_view.tsx

```diff
@@ -98,21 +101,32 @@ export const useDataView = (): { indexFieldsSearch: (selectedDataViewId: string)
               addError(msg, {
                 title: i18n.FAIL_BEAT_FIELDS,
               });
-              searchSubscription$.current.unsubscribe();
+              searchSubscription$.current[selectedDataViewId].unsubscribe();
             },
           });
+        searchSubscription$.current = {
```

Comment:
the unsubscribes in the above functions are called before the search is assigned to  `searchSubscription$.current[selectedDataViewId]`. blows up the tests